app/models/user.rb

```diff
@@ -1,8 +1,14 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable, :confirmable
   has_many :posts, foreign_key: 'author_id'
   has_many :comments, foreign_key: 'author_id'
   has_many :likes, foreign_key: 'author_id'
 
+  attribute :posts_counter, :integer, default: 0
+
   def self.recent_posts(name)
     p Post.where(author_id: User.find_by(name:).id).order(created_at: :desc).limit(3)
   end
```

Comment:
- [ ] On line 16 of this file, you added validation that insists that the User `:name` cannot be blank. Meanwhile, when you are creating a User by taking data from the user, no fiend receives a name to create a user. 

![image](https://user-images.githubusercontent.com/84156714/231317439-ee020c8e-72b3-4d18-adda-2f8343da9120.png)

I should be able to create a user but I get this error message. Kindly fix this.

Reply:
Because I could not create a user, I cannot test if `:confirmable`, `:recoverable`, `:rememberable`, and `:validatbale`. I am sure the next reviewer will check this once if they work as expected.